Description: Modify upstreams Makefiles to clean correctly
Author: Christian Welzel <gawain@camlann.de>
Last-Update: 2012-02-05

--- swftools.orig/Makefile.in
+++ swftools/Makefile.in
@@ -16,8 +16,6 @@
 	cd lib/python;$(MAKE) $@
 	@echo making $@ in lib/ruby...
 	cd src;$(MAKE) $@
-	@echo making $@ in avi2swf...
-	cd avi2swf;$(MAKE) $@
 	@echo making $@ in swfs...
 	cd swfs;$(MAKE) $@
 	@$(MAKE) $@-local
